pkg/controller/accountclaim/reuse.go

```diff
@@ -32,6 +34,45 @@ func (r *ReconcileAccountClaim) finalizeAccountClaim(reqLogger logr.Logger, acco
 		return err
 	}
 
+	// Region comes from accountClaim
+	clusterAwsRegion := accountClaim.Spec.Aws.Regions[0].Name
+	if reusedAccount.Spec.BYOC {
+		// AWS credential comes from accountclaim object osdCcsAdmin user
+		// We must use this user as we would other delete the osdManagedAdmin
+		// user that we're going to delete
+		// TODO: We should use the role here
+		awsClientInput = awsclient.NewAwsClientInput{
+			SecretName: accountClaim.Spec.BYOCSecretRef.Name,
+			NameSpace:  accountClaim.Namespace,
+			AwsRegion:  clusterAwsRegion,
+		}
+	} else {
+		// AWS credential comes from account object
+		awsClientInput = awsclient.NewAwsClientInput{
+			SecretName: reusedAccount.Spec.IAMUserSecret,
+			NameSpace:  awsv1alpha1.AccountCrNamespace,
+			AwsRegion:  clusterAwsRegion,
+		}
+	}
+
+	awsClient, err := awsclient.GetAWSClient(r.client, awsClientInput)
+
+	if err != nil {
+		connErr := fmt.Sprintf("Unable to create aws client for region %s", clusterAwsRegion)
+		reqLogger.Error(err, connErr)
+		return err
+	}
+
+	// Remove IAM user we'll remove the IAM user for CCS
+	if utils.AccountCRHasIAMUserIDLabel(reusedAccount) && accountClaim.Spec.BYOC {
+		err = r.cleanUpIAM(reqLogger, awsClient, reusedAccount, accountClaim)
+		if err != nil {
+			reqLogger.Error(err, "Failed to delete IAM user during finalizer cleanup")
```

Comment:
Do we want to return from the reconcile loop with error in case we fall here?

Reply:
No I don't think we want to fail the reuse for this since its CCS, these users are unique so leaving them behind isn't a problem